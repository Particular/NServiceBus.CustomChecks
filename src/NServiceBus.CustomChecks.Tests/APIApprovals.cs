using NServiceBus.CustomChecks;
using NUnit.Framework;
using Particular.Approvals;
using PublicApiGenerator;

[TestFixture]
public class APIApprovals
{
    [Test]
    public void Approve()
    {
        var publicApi = ApiGenerator.GeneratePublicApi(typeof(CustomCheck).Assembly);
        Approver.Verify(publicApi);
    }
}