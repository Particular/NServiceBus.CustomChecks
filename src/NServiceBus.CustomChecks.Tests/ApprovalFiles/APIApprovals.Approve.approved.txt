[assembly: System.Runtime.CompilerServices.InternalsVisibleToAttribute(@"NServiceBus.CustomChecks.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100dde965e6172e019ac82c2639ffe494dd2e7dd16347c34762a05732b492e110f2e4e2e1b5ef2d85c848ccfb671ee20a47c8d1376276708dc30a90ff1121b647ba3b7259a6bc383b2034938ef0e275b58b920375ac605076178123693c6c4f1331661a62eba28c249386855637780e3ff5f23a6d854700eaa6803ef48907513b92")]
namespace NServiceBus.CustomChecks
{
    public class CheckResult
    {
        public static NServiceBus.CustomChecks.CheckResult Pass;
        public CheckResult() { }
        public string FailureReason { get; set; }
        public bool HasFailed { get; set; }
        public static NServiceBus.CustomChecks.CheckResult Failed(string reason) { }
    }
    public abstract class CustomCheck : NServiceBus.CustomChecks.ICustomCheck
    {
        protected CustomCheck(string id, string category, System.Nullable<System.TimeSpan> repeatAfter = null) { }
        public string Category { get; }
        public string Id { get; }
        public System.Nullable<System.TimeSpan> Interval { get; }
        public abstract System.Threading.Tasks.Task<NServiceBus.CustomChecks.CheckResult> PerformCheck();
    }
    public interface ICustomCheck
    {
        string Category { get; }
        string Id { get; }
        System.Nullable<System.TimeSpan> Interval { get; }
        System.Threading.Tasks.Task<NServiceBus.CustomChecks.CheckResult> PerformCheck();
    }
}
namespace NServiceBus
{
    public class static CustomChecksConfigurationExtensions
    {
        public static void ReportCustomChecksTo(this NServiceBus.EndpointConfiguration config, string serviceControlQueue, System.Nullable<System.TimeSpan> timeToLive = null) { }
    }
}